### DISCUSSION TITLE
ErrorCheckNewFrameSanityChecks assertion failed

### DISCUSSION THREAD

**Discussion by roby2014** [ocornut/imgui#5323] [New Users Build/Link/Run issues ONLY!]
Hey, trying to run a simple program but I'm getting an assertion error.
```
/external/imgui/imgui.cpp:8607: void ImGui::ErrorCheckNewFrameSanityChecks(): Assertion `0 && "Please set DockingEnable before the first call to NewFrame()! Otherwise you will lose your .ini settings!"' failed.
[1]    6208 IOT instruction (core dumped)
```

Even tho it sounds weird, I'm only getting this error on Linux, Windows runs it fine.

This is the part that it crashes:
```cpp
auto& io = ImGui::
io.ConfigFlags |= 
ImGui::DockSpaceOverViewport(ImGui::GetMainViewport());
```

**I cant understand why, since I am actually setting `DockingEnable` before doing anything.**

*Full code in case its needed:*
```cpp
int main() {
    sf::RenderWindow window(sf::VideoMode(1200, 600), "ImGui + SFML = <3");
    window.setFramerateLimit(60);
    ImGui::SFML::Init(window);

    sf::CircleShape shape(100.f);
    shape.setFillColor(sf::Color::Green);

    sf::Clock deltaClock;
    while (window.isOpen()) {
        sf::Event event;
        while (window.pollEvent(event)) {
            ImGui::SFML::ProcessEvent(window, event);

            if (event.type == sf::Event::Closed) {
                window.close();
            }
        }

        ImGui::SFML::Update(window, deltaClock.restart());

        // CRASH
        auto& io = ImGui::GetIO();
        io.ConfigFlags |= ImGuiConfigFlags_DockingEnable;
        ImGui::DockSpaceOverViewport(ImGui::GetMainViewport());

        window.clear();
        window.draw(shape);
        ImGui::SFML::Render(window);
        window.display();
    }

    ImGui::SFML::Shutdown();
}
```

Thanks in advance.

**Comment by ocornut** [comment]
The assert exactly says what’s wrong:
“ Please set DockingEnable before the first call to NewFrame()!”

Not sure what else there is to add.

> I cant understand why, since I am actually setting DockingEnable before doing anything.

That’s not true in the code you posted.
You are calling ImGui::SFML::Update() which calls NewFrame(). I guess it is all a bit obfuscated by the fact the author of ImGui::SFML decided to ignore the naming convention of the 20+ other backends and call their NewFrame() function Update().

**Reply by roby2014** [reply]
Hi, thanks for answering.
You're right, I was just not understanding why would this crash on Linux but not on Windows, maybe I forgot something no idea. Code works just fine now.
Sorry, I'll pay more attention next time.
